File: Day-01/java/rechergg/Day1.java
package rechergg;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;

public class Day1 {

    private static final String[] NUMBERS = {"one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};

    public static void main(String[] args) throws IOException {
        String content = Files.readString(Path.of("./Day-01/java/rechergg/input.txt"));

        System.out.printf("Ergebnis aus Part (1): %d%n", content.lines().mapToInt(line -> findFirstNumber(line) + findLastNumber(line)).sum());

        for (int i = 0; i < NUMBERS.length; i++) {
            content = content.replaceAll(NUMBERS[i], NUMBERS[i] + (i + 1) + NUMBERS[i]);
        }

        System.out.printf("Ergebnis aus Part (2): %d%n", content.lines().mapToInt(line -> findFirstNumber(line) + findLastNumber(line)).sum());
    }

    public static int findFirstNumber(String line) {
        for (int i = 0; i < line.length(); i++) {
            int charAsNumber = Character.getNumericValue(line.charAt(i));
            if (charAsNumber >= 1 && charAsNumber <= 9) {
                return charAsNumber * 10;
            }
        }
        return 0;
    }

    public static int findLastNumber(String content) {
        for (int i = content.length() - 1; i >= 0; i--) {
            int charAsNumber = Character.getNumericValue(content.charAt(i));
            if (charAsNumber >= 1 && charAsNumber <= 9) {
                return charAsNumber;
            }
        }
        return 0;
    }
}

File: Day-01/typescript/Day1.ts
const fs = require('fs');

const NUMBERS = ["one", "two", "three", "four", "five", "six", "seven", "eight", "nine"];

function main() {
    let content = fs.readFileSync('../../input.txt', 'utf-8');

    console.log(`Ergebnis aus Part (1): ${content.split('\n').map(line => findFirstNumber(line) + findLastNumber(line)).reduce((acc, curr) => acc + curr, 0)}`);

    for (let i = 0; i < NUMBERS.length; i++) {
        const regex = new RegExp(NUMBERS[i], 'g');
        content = content.replace(regex, `${NUMBERS[i]}${i + 1}${NUMBERS[i]}`);
    }

    console.log(`Ergebnis aus Part (2): ${content.split('\n').map(line => findFirstNumber(line) + findLastNumber(line)).reduce((acc, curr) => acc + curr, 0)}`);
}

function findFirstNumber(line) {
    for (let i = 0; i < line.length; i++) {
        const charAsNumber = parseInt(line.charAt(i));
        if (!isNaN(charAsNumber) && charAsNumber >= 1 && charAsNumber <= 9) {
            return charAsNumber * 10;
        }
    }
    return 0;
}

function findLastNumber(line) {
    for (let i = line.length - 1; i >= 0; i--) {
        const charAsNumber = parseInt(line.charAt(i));
        if (!isNaN(charAsNumber) && charAsNumber >= 1 && charAsNumber <= 9) {
            return charAsNumber;
        }
    }
    return 0;
}

main();